src/systems/filecoin_mining/storage_mining/storage_miner_actor.go

```diff
@@ -543,44 +543,97 @@ func (st *StorageMinerActorState_I) _isSealVerificationCorrect(rt Runtime, onCha
 	return false // TODO: finish
 }
 
-func (st *StorageMinerActorState_I) _sectorExists(rt Runtime, sectorNo sector.SectorNumber) bool {
+func (st *StorageMinerActorState_I) _sectorExists(sectorNo sector.SectorNumber) bool {
 	_, found := st.Sectors()[sectorNo]
 	return found
 }
 
-// Currently deals must be posted on chain via sma.PublishStorageDeals before CommitSector
-// TODO(optimization): CommitSector could contain a list of deals that are not published yet.
-func (a *StorageMinerActorCode_I) CommitSector(rt Runtime, onChainInfo sector.OnChainSealVerifyInfo) {
+// Deals must be posted on chain via sma.PublishStorageDeals before PreCommitSector
+// TODO(optimization): PreCommitSector could contain a list of deals that are not published yet.
+func (a *StorageMinerActorCode_I) PreCommitSector(rt Runtime, info sector.SectorPreCommitInfo) {
+
+	// no checks needed
```

Comment:
do we need to validate any of `SectorPreCommitInfo` ? 
- ~~do we need to check dealIDs map to existing deals? probably worth checking here~~ there's a TODO below.
- is `SealRandomness` correct? do we need to check it now, or wait till `CommitSector`

Reply:
We might do some syntactic validation here but unclear how much we need to check in the PreCommit step. If that is the case, make sense to incur a cost for calling PreCommit